lib/fluent/plugin/formatter_syslog_rfc5424.rb

```diff
@@ -44,6 +47,14 @@ def format(tag, time, record)
 
         msg.length.to_s + ' ' + msg
       end
+
+      def get_structured_data(record)
+        sd_array = []
+        @structured_data_field_array.each do |sd_field|
+          sd_array << RFC5424::StructuredData.new(sd_id: record.dig(sd_field) ).to_s if record.dig(sd_field)
```

Comment:
I have mixed feeling about this. Basically why would you use dig here. I know they were using it before. But i want to know why before we copy that over. Maybe to handle the cases of default records for structured data being passed ?

Reply:
we need to create fluentd fields using record_transformer in flow and then we can use it in output as structured_data_field. so basically records contains all the key-value-pair and using dig we need find token value based upon structured_data_field.
```
 in output : 
structured_data_field => structured_data_66d4f2d0-ac54-4290-bedb-971a8deb5f24

in flow
- record_transformer:
       records:
       - structured_data_66d4f2d0-ac54-4290-bedb-971a8deb5f24: <token>
```